pandas/core/tools/datetimes.py

```diff
@@ -762,7 +762,9 @@ def to_datetime(
         If parsing succeeded.
         Return type depends on input:
 
-        - list-like: DatetimeIndex
+        - list-like:
+            - DatetimeIndex, if timezone naive or aware with the same timezone
```

Comment:
As discussed in https://github.com/pandas-dev/pandas/issues/42229#issuecomment-870009579

```suggestion
            - DatetimeIndex, if timezone naive or aware with the same time offset.
```